### PR TITLE
Upgrade to k8s v1.27

### PR DESCRIPTION
**Pull request by hjiawei**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by hjiawei** [comment]
UT/FVs are finally [green](https://tigera.semaphoreci.com/workflows/717df5a5-477e-4a9e-952d-cf09af50ef08?pipeline_id=0888c1b2-d83f-44b0-82bf-f70f2ce5e86f).

File: felix/aws/ec2.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ func WaitForEC2SrcDstCheckUpdate(check string, healthAgg *health.HealthAggregato
 		jitter        = 0.1
 	)
 
+	//nolint:staticcheck // Ignore SA1019 deprecated
```
Comment (coutinhop) [inline felix/aws/ec2.go:88]: should we have a FIXME comment here pointing out exactly what is deprecated and that we should remove it in the future?

File: felix/wireguard/bootstrap.go [hunk 2]
```diff
@@ -413,6 +413,7 @@ func removeWireguardForBootstrapping(
 
 // getPublicKeyForNode returns the configured wireguard public key for a given node.
 func getPublicKeyForNode(logCtx *log.Entry, nodeName string, calicoClient clientv3.NodesClient, maxRetries int, ipVersion uint8) (string, error) {
+	//nolint:staticcheck // Ignore SA1019 deprecated
```
Comment (coutinhop) [inline felix/wireguard/bootstrap.go:416]: same question about a FIXME comment

File: felix/wireguard/bootstrap.go [hunk 3]
```diff
@@ -522,6 +523,7 @@ func removeWireguardDevice(
 
 	logCtx.Debug("Removing wireguard device")
 
+	//nolint:staticcheck // Ignore SA1019 deprecated
```
Comment (coutinhop) [inline felix/wireguard/bootstrap.go:526]: same question about a FIXME comment

File: felix/wireguard/bootstrap.go [hunk 4]
```diff
@@ -570,6 +572,7 @@ func removeWireguardPublicKey(
 		return fmt.Errorf("unknown IP version: %d", ipVersion)
 	}
 
+	//nolint:staticcheck // Ignore SA1019 deprecated
```
Comment (coutinhop) [inline felix/wireguard/bootstrap.go:575]: same question about a FIXME comment

**Review by coutinhop — APPROVED** [review]
LGTM, only a few questions about adding more info when disabling lint

File: felix/aws/ec2.go (hunk 1 above)
Comment (hjiawei) [reply felix/aws/ec2.go:88]: Regarding the ignored staticcheck, `wait.NewExponentialBackoffManger` is [deprecated in k8s/apimachinery](https://github.com/kubernetes/apimachinery/blob/194b6f8ddf62f55d1b8d997223e94ace2c1def31/pkg/util/wait/backoff.go#L354) v1.27+. The suggested replacement is `wait.BackoffUntil` but [it still needs](https://github.com/kubernetes/apimachinery/blob/194b6f8ddf62f55d1b8d997223e94ace2c1def31/pkg/util/wait/backoff.go#L211) a `BackoffManager` as the second parameter. We can't really get away from the `NewExponentialBackoffManger` call to instantiate an instance. This issue is tracked internally and we will remove or fix this when the k8s dependency is updated.

Comment (coutinhop) [reply felix/aws/ec2.go:88]: thanks for clarifying @hjiawei! sounds good to me!

**Pull request merged**